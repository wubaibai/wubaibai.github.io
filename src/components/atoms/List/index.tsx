import React from 'react';

export interface Item {
	key: string;
	value: string;
}

interface ListProperty {
	items: Item[];
}

const List: React.FC<ListProperty> = ({ items }) => (
	<ul>
		{items.map(item => (
			<li key={item.key} data-testid={item.key}>
				{item.value}
			</li>
		))}
	</ul>
);

export default List;
