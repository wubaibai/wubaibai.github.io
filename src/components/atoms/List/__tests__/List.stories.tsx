import React from 'react';
import { storiesOf } from '@storybook/react';
import { object } from '@storybook/addon-knobs';

import List from 'components/atoms/List';

const items = [
	{
		key: '1',
		value: 'Lee',
	},
	{
		key: '2',
		value: 'Mike',
	},
];

const stories = storiesOf('atoms/List', module);

stories.add('__interactive', () => <List items={object('list', items)} />, { jest: 'List' });
