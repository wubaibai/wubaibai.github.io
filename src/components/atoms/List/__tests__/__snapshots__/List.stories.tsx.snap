// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <ul>
    <li
      data-testid="1"
    >
      Lee
    </li>
    <li
      data-testid="2"
    >
      Mike
    </li>
  </ul>
</DocumentFragment>
`;
